frontend/www/js/omegaup/components/user/UserHeatmap.vue

```diff
@@ -0,0 +1,233 @@
+<template>
+  <div class="user-heatmap-container">
+    <div class="user-heatmap-wrapper">
+      <div class="year-selector">
+        <select
+          v-model="selectedYear"
+          class="form-control"
+          @change="onYearChange"
+        >
+          <option v-for="year in availableYears" :key="year" :value="year">
+            {{ year }}
+          </option>
+        </select>
+      </div>
+      <div ref="heatmapContainer" class="user-heatmap"></div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop, Ref, Watch } from 'vue-property-decorator';
+import * as Highcharts from 'highcharts/highstock';
+import HighchartsHeatmap from 'highcharts/modules/heatmap';
+import T from '../../lang';
+import * as ui from '../../ui';
+import { getHeatmapChartOptions } from '../../user/profile';
+
+HighchartsHeatmap(Highcharts);
+
+interface HeatmapDataPoint {
+  date: string;
+  count: number;
+}
+
+// Define color variables
+export const COLORS = {
+  emptyCell: 'var(--user-heatmap-empty-cell-color)',
+  lowActivity: 'var(--user-heatmap-low-activity-color)',
+  mediumActivity: 'var(--user-heatmap-medium-activity-color)',
+  highActivity: 'var(--user-heatmap-high-activity-color)',
+  background: 'var(--user-heatmap-background-color)',
+  wrapper: 'var(--user-heatmap-wrapper-background-color)',
+};
+
+@Component({})
+export default class UserHeatmap extends Vue {
+  @Prop() username!: string;
+  @Prop({ default: () => [] }) heatmapData!: HeatmapDataPoint[];
+  @Prop({ default: () => [] }) availableYears!: number[];
+  @Prop({ default: false }) isLoading!: boolean;
+
+  chart: Highcharts.Chart | null = null;
+  selectedYear: number = new Date().getFullYear();
+  T = T;
+  ui = ui;
+  COLORS = COLORS;
+
+  @Ref('heatmapContainer') readonly heatmapContainer!: HTMLElement;
+
+  mounted(): void {
+    this.$nextTick(() => {
+      if (this.availableYears && this.availableYears.length > 0) {
+        this.selectedYear = this.availableYears[0];
+      }
+
+      if (this.heatmapData && this.heatmapData.length > 0) {
+        this.renderHeatmap();
+      }
+    });
+  }
+
+  @Watch('availableYears', { immediate: true, deep: true })
+  onAvailableYearsChange(newValue: number[]): void {
+    if (!newValue || !newValue.length) return;
```

Comment:
Thanks to ES6, we can simplify this expression using [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining):
```suggestion
    if (!newValue?.length) return;
```